docs/zkapps/tutorials/05-common-types-and-functions.mdx

```diff
@@ -294,123 +337,126 @@ If a program is too large to fit into these constraints, it can be broken up int
 
 You can use [Merkle trees](../snarkyjs-reference/classes/MerkleTree) to manage large amounts of data within a circuit. The power of Merkle trees is demonstrated in the [05-common-types-and-functions/src](https://github.com/o1-labs/docs2/tree/main/examples/zkapps/05-common-types-and-functions/src) reference project for this tutorial. See the [BasicMerkleTreeContract.ts](https://github.com/o1-labs/docs2/blob/main/examples/zkapps/05-common-types-and-functions/src/BasicMerkleTreeContract.ts) contract and [main.ts](https://github.com/o1-labs/docs2/blob/main/examples/zkapps/05-common-types-and-functions/src/main.ts) that demonstrates how contracts interact with Merkle trees and how to construct them.
 
-The first step is to import `MerkleTree`:
-
-```ts
-import {
-  ...
-  MerkleTree,
-  ...
-} from 'snarkyjs'
-```
-
 To create Merkle trees in your application:
 
-```ts
-const height = 20;
-const tree = new MerkleTree(height);
+```ts src/run.ts
+114 const height = 20;
+115 const tree = new MerkleTree(height);
+116 class MerkleWitness20 extends MerkleWitness(height) {}
 ```
 
 The height variable determines how many leaves are available to the application. For example, a height of 20 leads to a tree with `2^(20-1)`, or 524,288 leaves.
 
 Merkle trees in smart contracts are stored as the hash of the Merkle tree's root. Smart contract methods that update the Merkle root can take a _witness_ of the change as an argument. The [MerkleMapWitness](/zkapps/snarkyjs-reference/classes/MerkleMapWitness) represents the Merkle path to the data for which inclusion is being proved.
 
-A contract stores the root of a Merkle tree, where each leaf stores a number, and the smart contract has an `update` function that adds a number to the leaf. 
+A contract stores the root of a Merkle tree, where each leaf stores a number, and the smart contract has an `update` function that adds a number to the leaf.
 
 For example, to put a condition on a leaf update, the `update` function checks that the number added was less than 10:
 
-```ts
-...
-  @state(Field) treeRoot = State<Field>();
-...
-  @method initState(initialRoot: Field) {
-    this.treeRoot.set(initialRoot);
-  }
-
-  @method update(
-    leafWitness: MerkleWitness20,
-    numberBefore: Field,
-    incrementAmount: Field
-  ) {
-    const initialRoot = this.treeRoot.get();
-    this.treeRoot.assertEquals(initialRoot);
-
-    incrementAmount.assertLt(Field(10));
-
-    // check the initial state matches what we expect
-    const rootBefore = leafWitness.calculateRoot(numberBefore);
-    rootBefore.assertEquals(initialRoot);
-
-    // compute the root after incrementing
-    const rootAfter = leafWitness.calculateRoot(
-      numberBefore.add(incrementAmount)
-    );
-
-    // set the new root
-    this.treeRoot.set(rootAfter);
-  }
+```ts src/run.ts
+117 class BasicMerkleTreeContract extends SmartContract {
```

Comment:
hmm, so having the dots `...` as a way to tell the tutorial runner "skip any number of lines" would actually be useful. doesn't have to be in this PR but the sooner the better before we have to change all tutorials just because we don't have this feature